src/MatrixClient.cc

```diff
@@ -711,16 +711,16 @@ MatrixClient::uploadImage(const QString &roomid, const QString &filename)
                         return;
                 }
 
-                emit imageUploaded(roomid, filename, object.value("content_uri").toString());
+                emit imageUploaded(roomid, iodev.dynamicCast<QFile>(), object.value("content_uri").toString());
```

Comment:
It's unclear whether there's any handling for the case where this cast fails and returns an empty `QSharedPointer`. Regardless, as discussed elsewhere, `QFile` should not be used here.